tracing/opentelemetry-tracing-sync/README.md

```diff
@@ -0,0 +1,159 @@
+- Header attributes mapping
+
+## Usage
+
+### Basic Configuration
+
+```
+scalaCopier le codeimport sttp.tapir.server.opentelemetry._
+import io.opentelemetry.api.trace.Tracer
+
+// Get your OpenTelemetry tracer instance
+val tracer: Tracer = // ... your OTel configuration
+
+// Basic setup
+val tracing = new OpenTelemetryTracingSync(tracer)
+
+// Server integration
+val serverInterpreter = ServerInterpreter(tracing)
+```
+
+### Custom Configuration
+
+```
+scalaCopier le codeval config = OpenTelemetryConfig(
+  includeHeaders = Set("x-request-id", "user-agent"),
+  includeBaggage = true,
+  errorPredicate = _ >= 500,
+  spanNaming = SpanNaming.Path
+)
+
+val customTracing = new OpenTelemetryTracingSync(tracer, config)
+```
+
+### Span Naming Strategies
+
+```
+scalaCopier le code// Default: "METHOD /path"
+spanNaming = SpanNaming.Default
+
+// Path only: "/path"
+spanNaming = SpanNaming.Path
+
+// Custom naming
+spanNaming = SpanNaming.Custom(endpoint => s"API-${endpoint.showShort}")
+```
+
+## Configuration Options
+
+### OpenTelemetryConfig
+
+| Option           | Type                  | Default                 | Description                                      |
+| ---------------- | --------------------- | ----------------------- | ------------------------------------------------ |
+| `includeHeaders` | `Set[String]`         | `Set.empty`             | HTTP headers to include as span attributes       |
+| `includeBaggage` | `Boolean`             | `true`                  | Enable/disable OpenTelemetry baggage propagation |
+| `errorPredicate` | `Int => Boolean`      | `_ >= 500`              | Predicate to determine error status codes        |
+| `spanNaming`     | `SpanNaming`          | `Default`               | Strategy for naming spans                        |
+| `virtualThreads` | `VirtualThreadConfig` | `VirtualThreadConfig()` | Virtual threads specific options                 |
+
+### VirtualThreadConfig
+
+| Option                    | Type      | Default       | Description                     |
+| ------------------------- | --------- | ------------- | ------------------------------- |
+| `useVirtualThreads`       | `Boolean` | `true`        | Enable virtual threads usage    |
+| `virtualThreadNamePrefix` | `String`  | `"tapir-ot-"` | Prefix for virtual thread names |
+
+## Virtual Threads Compatibility
+
+This module is designed to work efficiently with Project Loom's virtual threads:
+
+- Uses `ScopedValue` instead of `ThreadLocal`
+- Proper context propagation across thread boundaries
+- Optimized for high-concurrency scenarios
+
+## Examples
+
+### Basic Server Setup
+
+```
+scalaCopier le codeimport sttp.tapir.server.opentelemetry._
+import io.opentelemetry.api.trace.Tracer
+
+def setupServer(tracer: Tracer) = {
+  val tracing = new OpenTelemetryTracingSync(tracer)
+  
+  val endpoint = endpoint.get
+    .in("hello")
+    .out(stringBody)
+    .serverLogic(_ => Right("Hello, World!"))
+
+  ServerInterpreter(tracing)

```

Comment:
it would be best to provide examples basing on a real server interpretr, netty-server-sync should be the best candidate (https://tapir.softwaremill.com/en/latest/server/netty.html)